### ISSUE TITLE
Problems compiling Pykep on OSX

### ISSUE THREAD

**Issue by NigelPB** [esa/pykep#66]
Hello,

I'm hitting problems compiling Pykep on OSX; I've seen someone reporting similar looking error messages but the error in that case seemed to be inconsistent Python version information in cmake. I don't think that's the case here, but I'm not an expert in cmake; can anyone suggest what the issue is?

Summary: I have installed boost and boost-python version 1.64 using Homebrew. This seems to be picked up OK in cmake. I'm using OSX 10.12.4, with Xcode V8.0. I downloaded the Pykep source using git clone.

I've tried the OSX "native" Python 2.7, and also a separate Python 2.7 installed via Anaconda - same results.

Here is the output from my cmake configuration:

```
-- OS detected: Darwin
-- CXX Compiler detected: AppleClang
-- CMake additional search path for libraries: /usr/local/lib
-- Enabling '-std=c++11' compiler flag
-- Enabling '-fPIC' compiler flag
-- CXX compilation flags:  -std=c++11
-- C compilation flags:  -fPIC
-- Python interpreter: /Users/nigelbannister/anaconda/bin/python
-- Python interpreter verison: 2.7
-- Python includes path: /Users/nigelbannister/anaconda/include/python2.7
-- Python modules install path: /Users/nigelbannister/anaconda/lib/python2.7/site-packages
-- Python library name: /Users/nigelbannister/anaconda/lib/libpython2.7.dylib
-- Required Boost libraries: serialization;date_time;python
-- Boost version: 1.64.0
-- Found the following Boost libraries:
--   serialization
--   date_time
--   python
-- Detected Boost version: 106400
-- Boost include dirs: /usr/local/include
-- Boost libraries: /usr/local/lib/libboost_serialization-mt.dylib;/usr/local/lib/libboost_date_time-mt.dylib;/usr/local/lib/libboost_python-mt.dylib
-- Creating the files for the generation of a manylinux binary wheel.
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/nigelbannister/pykep
```

When I then run make, I receive a lot of output containing numerous errors including "no member named 'serialize' and others. I've copied a section of the output below.

```
host-163-147:build nigelbannister$ make
[  0%] Building CXX object src/CMakeFiles/keplerian_toolbox_static.dir/planet/keplerian.cpp.o
In file included from /Users/nigelbannister/pykep/src/planet/keplerian.cpp:30:
In file included from /Users/nigelbannister/pykep/src/planet/keplerian.h:32:
In file included from /Users/nigelbannister/pykep/src/planet/base.h:32:
In file included from /Users/nigelbannister/pykep/src/planet/../epoch.h:35:
In file included from /Users/nigelbannister/pykep/src/planet/../serialization.h:30:
In file included from /usr/local/include/boost/archive/binary_iarchive.hpp:20:
In file included from /usr/local/include/boost/archive/binary_iarchive_impl.hpp:20:
In file included from /usr/local/include/boost/archive/basic_binary_iprimitive.hpp:53:
In file included from /usr/local/include/boost/serialization/array_wrapper.hpp:19:
In file included from /usr/local/include/boost/serialization/nvp.hpp:26:
In file included from /usr/local/include/boost/serialization/split_member.hpp:23:
/usr/local/include/boost/serialization/access.hpp:116:11: error: no member named 'serialize' in 'boost::array<double, 3>'
        t.serialize(ar, file_version);
        ~ ^
/usr/local/include/boost/serialization/serialization.hpp:68:13: note: in instantiation of function template specialization
      'boost::serialization::access::serialize<boost::archive::binary_iarchive, boost::array<double, 3> >' requested here
    access::serialize(ar, t, static_cast<unsigned int>(file_version));
            ^
/usr/local/include/boost/serialization/serialization.hpp:126:5: note: in instantiation of function template specialization
      'boost::serialization::serialize<boost::archive::binary_iarchive, boost::array<double, 3> >' requested here
    serialize(ar, t, v);
    ^
/usr/local/include/boost/archive/detail/iserializer.hpp:188:27: note: in instantiation of function template specialization
      'boost::serialization::serialize_adl<boost::archive::binary_iarchive, boost::array<double, 3> >' requested here
    boost::serialization::serialize_adl(
                          ^
/usr/local/include/boost/serialization/singleton.hpp:136:47: note: in instantiation of member function
      'boost::archive::detail::iserializer<boost::archive::binary_iarchive, boost::array<double, 3> >::load_object_data' requested here
        static detail::singleton_wrapper< T > t;
                                              ^
/usr/local/include/boost/serialization/singleton.hpp:149:16: note: in instantiation of member function
      'boost::serialization::singleton<boost::archive::detail::iserializer<boost::archive::binary_iarchive, boost::array<double, 3> >
      >::get_instance' requested here
        return get_instance();
               ^
/usr/local/include/boost/archive/detail/iserializer.hpp:414:20: note: (skipping 18 contexts in backtrace; use
      -ftemplate-backtrace-limit=0 to see all)
                >::get_const_instance()
                   ^
/usr/local/include/boost/serialization/export.hpp:66:12: note: in instantiation of member function
      'boost::serialization::singleton<boost::archive::detail::pointer_iserializer<boost::archive::binary_iarchive,
      kep_toolbox::planet::keplerian> >::get_const_instance' requested here
        >::get_const_instance();
           ^
/usr/local/include/boost/serialization/export.hpp:105:40: note: in instantiation of member function
      'boost::archive::detail::export_impl<boost::archive::binary_iarchive, kep_toolbox::planet::keplerian>::enable_load' requested here
    export_impl<Archive,Serializable>::enable_load(
                                       ^
/usr/local/include/boost/serialization/export.hpp:92:37: note: in instantiation of member function
      'boost::archive::detail::ptr_serialization_support<boost::archive::binary_iarchive, kep_toolbox::planet::keplerian>::instantiate'
      requested here
        &ptr_serialization_support::instantiate
                                    ^
/usr/local/include/boost/serialization/export.hpp:133:9: note: in instantiation of member function
      'boost::archive::detail::extra_detail::guid_initializer<kep_toolbox::planet::keplerian>::export_guid' requested here
        export_guid(boost::serialization::is_abstract< T >());
        ^
/Users/nigelbannister/pykep/src/planet/keplerian.cpp:169:1: note: in instantiation of member function
      'boost::archive::detail::extra_detail::guid_initializer<kep_toolbox::planet::keplerian>::export_guid' requested here
BOOST_CLASS_EXPORT_IMPLEMENT(kep_toolbox::planet::keplerian);
^
/usr/local/include/boost/serialization/export.hpp:158:35: note: expanded from macro 'BOOST_CLASS_EXPORT_IMPLEMENT'
        >::get_mutable_instance().export_guid();             \
                                  ^
In file included from /Users/nigelbannister/pykep/src/planet/keplerian.cpp:30:
In file included from /Users/nigelbannister/pykep/src/planet/keplerian.h:32:
In file included from /Users/nigelbannister/pykep/src/planet/base.h:32:
In file included from /Users/nigelbannister/pykep/src/planet/../epoch.h:35:
In file included from /Users/nigelbannister/pykep/src/planet/../serialization.h:30:
In file included from /usr/local/include/boost/archive/binary_iarchive.hpp:20:
In file included from /usr/local/include/boost/archive/binary_iarchive_impl.hpp:20:
In file included from /usr/local/include/boost/archive/basic_binary_iprimitive.hpp:53:
In file included from /usr/local/include/boost/serialization/array_wrapper.hpp:19:
In file included from /usr/local/include/boost/serialization/nvp.hpp:26:
In file included from /usr/local/include/boost/serialization/split_member.hpp:23:
/usr/local/include/boost/serialization/access.hpp:116:11: error: no member named 'serialize' in 'boost::array<double, 6>'
        t.serialize(ar, file_version);
```

There's much more output afterwards but it all seems to be complaining about the same basic issues. 

Any pointers to potential solutions would be welcome!

Regards

Nigel

**Comment by darioizzo** [comment]
Since you are linking to the conda python you should also use the boost
libraries from conda and compiling pykep with clang++ .... that is
potentially the issue you are having.

**Comment by NigelPB** [comment]
Thanks for your quick reply Dario; I did that (uninstalled the Brew version of boost, installed Anaconda version), and it seems to have fixed the issue. I needed to do a couple of other things as well:

1) I got some boost/.../.hpp file not found errors during make. So instead of doing make and sudo make install from the /Pykep/build directory which is the advice on the Pykep installation page, I found I had to go up one level (to /Pykep) - then the errors were cleared.

2) I deleted the version of PyGMO installed through Anaconda, and built the version from git instead. 

All now seems to be fine, and I can run the Pykep examples no problem.

Thanks for your help,

Nigel
